### PR TITLE
Feature/application entity

### PR DESCRIPTION
**Pull request by ranjeet1515**
Added database entities.

---
This change is [Reviewable](https://reviewable.io/reviews/yugandhargangu/bettle-track/12)

### CODE REVIEW

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/common/AppSettings.java~ [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.bettle.track.entities.common;
```
Comment (shashankitmaster) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/common/AppSettings.java~]: Please remove this class

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Issue.java~ [hunk 2]
```diff
@@ -0,0 +1,99 @@
+package com.bettle.track.entities.issue;
```
Comment (shashankitmaster) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Issue.java~]: Please remove this class

**Review by shashankitmaster — CHANGES_REQUESTED** [review]
Please remove .keep files from repos.

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/User.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "unblocked_by", cascade = CascadeType.PERSIST)
+    @Filter(name = "active_ip_block")
+    private Set<IPBlock> listIPBlocks;
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "user_id", cascade = CascadeType.PERSIST)
+    @Filter(name = "active_dashboard")
+    private Set<Dashboard> listDashboards;
+
+    public Set<Dashboard> getListDashboards() {
```
Comment (shashankitmaster) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/User.java]: Remove this getter & setter  if you are using `@Data` in your class.

**Review by shashankitmaster — CHANGES_REQUESTED** [review]

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Category.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_categories")
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class Category extends AbstractParentEntity {
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Category.java]: Add @FilterDefs for active categories.

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Issue.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_issues")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Issue.java]: Change @FilterDef name from "issue" to "active_issues"

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueAttchment.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_attachment", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueAttchment.java]: Change @FilterDef name from "issue_attachment" to "active_issue_attachments".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueCategory.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_issues_categories")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issues_categories", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueCategory.java]: Change @FilterDef name from "issues_categories" to "active_issues_categories".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueComment.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_comments")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_comments", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueComment.java]: Change @FilterDef from "issue_comments" to "active_issue_comments".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueDesc.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_description")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_desc", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueDesc.java]: Change @FIlterDef name from "issue_desc" to "active_issue_desc"

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueField.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_issue_custom_fields")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_field", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueField.java]: Change @FilterDef name from "issue_field" to "active_issue_fields".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilter.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_filters")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "Issue_filter", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilter.java]: Change @FilterDef name from "Issue_filter" to "active_issue_filters".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilterField.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_filters")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "Issue_filter_field", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilterField.java]: Change @FilterDef name from "Issue_filter_field" to "active_issue_filter_fields".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilterSharing.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_filter_share")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "Issue_filter_sharing", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilterSharing.java]: Change @FilterDef name from "Issue_filter_sharing" to "active_issue_filter_sharing".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueHistory.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_issues_history")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_history", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueHistory.java]: Change @FilterDef name from "issue_history" to "active_issue_history".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueProcedure.java [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_procedure")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_procedures", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueProcedure.java]: Change @FilterDef name from "issue_procedures" to "active_issue_procedures".

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Tag.java [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+ * @author Name  : Ranjeet
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_tags")
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class Tag extends AbstractParentEntity {
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Tag.java]: Add @FilterDefs to find active tag records.

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/project/Project.java [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+ * @author Ranjeet Parashar
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_projects")
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class Project extends AbstractParentEntity {
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/project/Project.java]: Add @FilterDefs to find active projects.

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/User.java [hunk 18]
```diff
@@ -0,0 +1,192 @@
...
+ * @author Ranjeet Parashar
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_users")
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class User extends AbstractParentEntity {
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/User.java]: Add @FilterDefs for active users.

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/UserGroup.java [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+ * @author Ranjeet Parashar
+ * @version 1.0
+ * @since 1.0
+ */
+@Entity
+@Table(name = "m_user_group")
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class UserGroup extends AbstractParentEntity {
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/UserGroup.java]: Add @FilterDefs for active user groups.

File: application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/UserInfo.java [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+ * Create Date: 2017/01/27
+ *
+ * @author Ranjeet Parashar
+ * @version 1.0
+ * @since 1.0
+ */
+
+@Entity
+@Table(name = "m_user_info")
+@FilterDef(name = "active_user_info", defaultCondition = "active_flag = 1")
```
Comment (yugandhargangu) [inline application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/UserInfo.java]: Add @FilterDefs and keep @FilterDef in @FilterDefs as array.
Add @Where above @FilterDefs annotation.

**Comment by shashankitmaster** [comment]
@ranjeet1515 : Please add package-info.java file for your package.

**Comment by ranjeet1515** [comment]
Done with all changes.

**Review by shashankitmaster — APPROVED** [review]

**Review by shashankitmaster — CHANGES_REQUESTED** [review]
Please change all `application.properties` to `application.yml`

**Review by shashankitmaster — DISMISSED** [review]
Add `sourceSet` to `build.gradle`

```java
sourceSets {
    main {
        java {
            srcDirs = ['src/java']
        }
        resources {
            srcDirs = ['src/resources']
        }
    }
}
```

**Comment by yugandhargangu** [comment]
Reviewed 21 of 54 files at r2, 40 of 40 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/yugandhargangu/bettle-track/12)*

**Review by shashankitmaster — APPROVED** [review]

**Comment by shashankitmaster** [comment]
Reviewed 21 of 54 files at r2, 40 of 40 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/yugandhargangu/bettle-track/12)*

**Pull request merged**